### PR TITLE
Structural patterns

### PR DESCRIPTION
**Pull request by chrispyles**
Adds structural patterns for matching complex Python objects.

### CODE REVIEW

**Comment by chrispyles** [comment]
FYI @leestott

**Comment by chrispyles** [comment]
@leestott added the usage info to the docs. also it looks like I no longer have write access to the repo so I can't publish a release. Once you merge this can you run the action w/ version 0.7.0? I already updated the changelog.

**Review by leestott — APPROVED** [review]
Great updates also thanks for validation of the structural function as suggested by Anthony and validating that this will not work as it provides boolean responses. Good updates to the docs also.

**Pull request merged**
